src/ripple/core/impl/DatabaseCon.cpp

```diff
@@ -38,9 +40,119 @@ setup_DatabaseCon(Config const& c)
         Throw<std::runtime_error>("database_path must be set.");
     }
 
+    if (!setup.globalPragma)
+    {
+        setup.globalPragma = [&c, &j]() {
+            auto const& sqlite = c.section("sqlite");
+            auto result = std::make_unique<std::vector<std::string>>();
+            result->reserve(3);
+
+            // defaults
+            std::string safety_level = "high";
```

Comment:
I disagree, but I think it's because of how I'm viewing the settings overall. I think of the `safety_level` as a baseline of settings, then all the others as overrides of the `safety_level`. In other words, the default is "high", but if I want transactions to be asyncronous, I can set `synchronous=off`. Similarly, if I want the lower safety level, but want to make sure that logging is safer, I could set
```
safety_level=low
journal_mode=wal
```

It's entirely possible I'm overthinking this. I could update the documentation in the example config file to be clearer that safety_level is the baseline and the others override.

> I think part of what you're doing with showWarning addresses this

The intention of `showWarning` is specific to when the config is set up to store a _lot_ of history. Think full history node, but since it's just a warning message, I set the cutoff lower.

Reply:
> Why would anybody choose only 1 or 2, and not all 3?

Excellent question. Some of the options can actually lower the data integrity risk, at the cost of extra I/O. I figured if I'm going to provide the options, I should go all the way, since I'm sure there are some folks who would use them.

> Perhaps safety_level should be removed

The more I think about it, the more I think it makes sense to make `safety_level` mutually exclusive from the other options. Someone who wants a simple option can use `safety_level=low`, and someone who wants to tune with more detail can whatever combos of the other three that they want. In other words, the example I gave above is starting to feel contrived and unrealistic.

Reply:
I agree.